src/Commands/MeltorCommand.php

```diff
@@ -0,0 +1,309 @@
+     *
+     * Returns true success, but also if there are still changes to review.
+     * Returns false on complete failures.
+     *
+     * @return bool
+     */
+    protected function testrun(): bool
+    {
+        $success                 = false;
+        $beforeStructureFileName = $this->meltor->combinePath(
+            $this->meltor->config('testrun.folder'),
+            $this->meltor->config('testrun.beforeStructureFileName')
+        );
+        $afterStructureFileName  = $this->meltor->combinePath(
+            $this->meltor->config('testrun.folder'),
+            $this->meltor->config('testrun.afterStructureFileName')
+        );
+
+        // Reading existing database structure.
+        $this->newLine();
+        $this->line('Performing structure comparison test run');
+        $this->meltor->writeStructureDump($beforeStructureFileName, $this);
+
+        // Backup existing database.
+        $this->newLine();
+        $this->line('Backing up database before applying temporary changes');
+        $this->call('protector:export', ['--file' => $this->meltor->config('testrun.backupFileName')]);
+
+        // Try out new migration.
+        $this->newLine();
+        $this->line('Running consolidated migration');
+        $this->call('migrate:fresh', []);
+
+        // Reading structure generated by the new migration.
+        $this->meltor->writeStructureDump($afterStructureFileName, $this);
+
+        // Compare both structures to make sure the new, consolidated migration does what all the previous ones did.
+        $this->newLine();
+        $this->line('Comparing structure dumps');
+
+        $fileSizeBefore  = filesize($beforeStructureFileName);
+        $fileSizeAfter   = filesize($afterStructureFileName);
+        $beforeStructure = $this->meltor->readAndCleanStructure($beforeStructureFileName);
+        $afterStructure  = $this->meltor->readAndCleanStructure($afterStructureFileName);
+
+        file_put_contents($beforeStructureFileName, $beforeStructure);
+        file_put_contents($afterStructureFileName, $afterStructure);
+
+        if (!$fileSizeBefore || !$fileSizeAfter) {
+            $this->warn('At least one structure file is empty, that\'s not good!');
+        } elseif ($beforeStructure == $afterStructure) {
+            $this->info('Success! the structure files are identical!');
+            $success = true;
+        } else {
+            $this->warn('The structure files are not identical, please check!');
+            $success = true;
+        }
+
+        $this->line($beforeStructureFileName);
+
+        if (!$fileSizeBefore) {
+            $this->warn('Something went wrong, the before file is empty!');
+        }
+
+        $this->line($afterStructureFileName);
+
+        if (!$fileSizeAfter) {
+            $this->warn('Something went wrong, the after file is empty!');
+        }
+
+        // Restore existing database.
+        $this->meltor->restoreBackup($this);
```

Comment:
If you have a restoreBackup method, maybe line 260 should also be wrapped in a createBackup method?